Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,384 @@
+
+            if (uri.EndsWith(".gltf"))
+            {
+                string gltfJson = File.ReadAllText(uri);
+
+                gltfObject = GetGltfObjectFromJson(gltfJson);
+
+                if (gltfObject == null)
+                {
+                    Debug.LogError("Failed load Gltf Object from json schema.");
+                    return null;
+                }
+            }
+            else if (uri.EndsWith(".glb"))
+            {
+                isGlb = true;
+                byte[] glbData;
+
+#if WINDOWS_UWP
+
+                if (loadAsynchronously)
+                {
+                    try
+                    {
+                        var storageFile = await Windows.Storage.StorageFile.GetFileFromPathAsync(uri);
+
+                        if (storageFile == null)
+                        {
+                            Debug.LogError($"Failed to locate .glb file at {uri}");
+                            return null;
+                        }
+
+                        var buffer = await Windows.Storage.FileIO.ReadBufferAsync(storageFile);
+
+                        using (Windows.Storage.Streams.DataReader dataReader = Windows.Storage.Streams.DataReader.FromBuffer(buffer))
+                        {
+                            glbData = new byte[buffer.Length];
+                            dataReader.ReadBytes(glbData);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Debug.LogError(e.Message);
+                        return null;
+                    }
+                }
+                else
+                {
+                    glbData = UnityEngine.Windows.File.ReadAllBytes(uri);
+                }
+#else
+                using (FileStream stream = File.Open(uri, FileMode.Open))
+                {
+                    glbData = new byte[stream.Length];
+
+                    if (loadAsynchronously)
+                    {
+                        await stream.ReadAsync(glbData, 0, (int)stream.Length);
+                    }
+                    else
+                    {
+                        stream.Read(glbData, 0, (int)stream.Length);
+                    }
+                }
+#endif
+
+                gltfObject = GetGltfObjectFromGlb(glbData);
+
+                if (gltfObject == null)
+                {
+                    Debug.LogError("Failed to load GlTF Object from .glb!");
+                    return null;
+                }
+            }
+            else
+            {
+                Debug.LogError("Unsupported file name extension.");
+                return null;
+            }
+
+            gltfObject.Uri = uri;
+            int nameStart = uri.Replace("\\", "/").LastIndexOf("/", StringComparison.Ordinal) + 1;
+            int nameLength = uri.Length - nameStart;
+            gltfObject.Name = uri.Substring(nameStart, nameLength).Replace(isGlb ? ".glb" : ".gltf", string.Empty);
+
```

Comment:
Might be possible to use a path API that does this:

[https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?view=netframework-4.7.2](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?view=netframework-4.7.2)

It looks like it exists in the .NET versions we care about.

Reply:
seems to work. Removes some flags/seems good